crates/swc_ecma_minifier/src/compress/pure/misc.rs

```diff
@@ -230,12 +230,172 @@ impl Pure<'_> {
         }
     }
 
+    /// `new RegExp("([Sap]+)", "ig")` => `/([Sap]+)/gi`
+    fn optimize_regex(&mut self, args: &mut Vec<ExprOrSpread>, span: &mut Span) -> Option<Expr> {
+        if args.is_empty() || args.len() > 2 {
+            return None;
+        }
+
+        // We aborts the method if arguments are not literals.
+        if args.iter().any(|v| {
+            v.spread.is_some()
+                || match &*v.expr {
+                    Expr::Lit(Lit::Str(s)) => {
+                        if s.value.contains(|c: char| {
+                            // whitelist
+                            !c.is_ascii_alphanumeric()
+                                && !matches!(c, '%' | '[' | ']' | '(' | ')' | '{' | '}' | '-' | '+')
+                        }) {
+                            return true;
+                        }
+                        if s.value.contains("\\\0") || s.value.contains('/') {
+                            return true;
+                        }
+
+                        false
+                    }
+                    _ => true,
+                }
+        }) {
+            return None;
+        }
+
+        let pattern = args[0].expr.take();
+
+        let pattern = match *pattern {
+            Expr::Lit(Lit::Str(s)) => s.value,
+            _ => {
+                unreachable!()
+            }
+        };
+
+        if pattern.is_empty() {
+            // For some expressions `RegExp()` and `RegExp("")`
+            // Theoretically we can use `/(?:)/` to achieve shorter code
+            // But some browsers released in 2015 don't support them yet.
+            args[0].expr = pattern.into();
+            return None;
+        }
+
+        let flags = args
+            .get_mut(1)
+            .map(|v| v.expr.take())
+            .map(|v| match *v {
+                Expr::Lit(Lit::Str(s)) => {
+                    assert!(s.value.is_ascii());
+
+                    let s = s.value.to_string();
+                    let mut bytes = s.into_bytes();
+                    bytes.sort_unstable();
+
+                    String::from_utf8(bytes).unwrap().into()
+                }
+                _ => {
+                    unreachable!()
+                }
+            })
+            .unwrap_or(js_word!(""));
+
+        Some(Expr::Lit(Lit::Regex(Regex {
+            span: *span,
+            exp: pattern,
+            flags,
+        })))
+    }
+
+    /// Array() -> []
+    fn optimize_array(&mut self, args: &mut Vec<ExprOrSpread>, span: &mut Span) -> Option<Expr> {
+        if args.len() == 1 {
+            if let ExprOrSpread { spread: None, expr } = &args[0] {
+                match &**expr {
+                    Expr::Lit(Lit::Num(num)) => {
+                        if num.value <= 11_f64 && num.value >= 0_f64 {
```

Comment:
why `11`?
It seems that 0 ~ 6 is reasonable.

Reply:
https://github.com/terser/terser/blob/master/lib/compress/index.js#L2086